### PR TITLE
Fix Minecraft Launcher runtime detection on Windows and macOS

### LINKED ISSUE #2332: Can't detect java bundled with launcher on mac
### Operating System

_No response_

### Version of Prism Launcher

Prism Launcher 8.2

### Version of Qt

Qt 6.6.2

### Description of bug

According to the pr there should be a path pointing to the java bundled with the minecraft launcher

### Steps to reproduce

Open minecraft launcher.
Launch game. 
detect java on prism (before or/and after game launch)
No path shows

### Suspected cause

Mac issue?

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### PR DESCRIPTION
**Pull request by Trial97**
fixes #2332

### CODE REVIEW

**Review by timoreo22 — APPROVED** [review]

File: launcher/java/JavaUtils.cpp [hunk 1]
```diff
@@ -441,13 +441,13 @@ QString JavaUtils::getJavaCheckPath()
...
 {
-    QString partialPath;
     QString executable = "java";
     QStringList processpaths;
 #if defined(Q_OS_OSX)
-    partialPath = FS::PathCombine(QDir::homePath(), "Library/Application Support");
+    processpaths << FS::PathCombine(QDir::homePath(), FS::PathCombine("Library", "Application Support", "minecraft", "runtime"));
 #elif defined(Q_OS_WIN32)
-    partialPath = QProcessEnvironment::systemEnvironment().value("LOCALAPPDATA", "");
+    QString partialPath = QProcessEnvironment::systemEnvironment().value("LOCALAPPDATA", "");
```
Comment (TheKodeToad) [inline launcher/java/JavaUtils.cpp]: Why are we using LOCALAPPDATA
That would make it AppData\Local\\.minecraft when it is located in AppData\Roaming\\.minecraft

Comment (Trial97) [reply launcher/java/JavaUtils.cpp]: You are right I should not need to check `AppData\Roaming\.minecraft\runtimes` folder(most likely doesn't exist)
But I use the AppData\Roaming prefix to locate where the Minecraft stores the java runtimes

Comment (TheKodeToad) [reply launcher/java/JavaUtils.cpp]: huh? that does exist for the non ms store launcher
it's just it was detecting it from AppData\Local which never exists

File: launcher/java/JavaUtils.cpp [hunk 2]
```diff
@@ -441,26 +441,25 @@ QString JavaUtils::getJavaCheckPath()
...
+    processpaths << FS::PathCombine(QFileInfo(appDataPath).absolutePath(), ".minecraft", "runtime");
+
     // add the microsoft store version of the launcher to the search. the current path is:
     // C:\Users\USERNAME\AppData\Local\Packages\Microsoft.4297127D64EC6_8wekyb3d8bbwe\LocalCache\Local\runtime
+    auto localAppDataPath = QProcessEnvironment::systemEnvironment().value("LOCALAPPDATA", "");
     auto minecraftMSStorePath =
-        FS::PathCombine(QFileInfo(partialPath).absolutePath(), "Local", "Packages", "Microsoft.4297127D64EC6_8wekyb3d8bbwe");
-    minecraftMSStorePath = FS::PathCombine(minecraftMSStorePath, "LocalCache", "Local", "runtime");
-    processpaths << minecraftMSStorePath;
+        FS::PathCombine(QFileInfo(localAppDataPath).absolutePath(), "Local", "Packages", "Microsoft.4297127D64EC6_8wekyb3d8bbwe");
```
Comment (getchoo) [inline launcher/java/JavaUtils.cpp]:
```suggestion
        FS::PathCombine(QFileInfo(localAppDataPath).absolutePath(), "Packages", "Microsoft.4297127D64EC6_8wekyb3d8bbwe");
```

**Review by getchoo — CHANGES_REQUESTED** [review]

**Review by getchoo — APPROVED** [review]

**Review by TheKodeToad — APPROVED** [review]

**Comment by Testsr** [comment]
It seems like this change has broken my Minecraft Launcher runtime detection. 8.2 works fine.

**Pull request merged**